config/dev/openstack-credentials.yaml

```diff
@@ -22,10 +24,85 @@ kind: Credential
 metadata:
   name: openstack-cluster-identity-cred
   namespace: ${NAMESPACE}
+  labels:
+    k0rdent.mirantis.com/component: "kcm"
 spec:
   description: OpenStack credentials
   identityRef:
     apiVersion: v1
     kind: Secret
     name: openstack-cloud-config
     namespace: ${NAMESPACE}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: openstack-cloud-config-resource-template
+  namespace: ${NAMESPACE}
+  labels:
+    k0rdent.mirantis.com/component: "kcm"
+  annotations:
+    projectsveltos.io/template: "true"
+data:
+  configmap.yaml: |
+    {{- $$cluster := .InfrastructureProvider -}}
+    {{- $$identity := (getResource "InfrastructureProviderIdentity") -}}
+
+    {{- $$clouds := fromYaml (index $$identity "data" "clouds.yaml" | b64dec) -}}
+    {{- if not $$clouds }}
+      {{ fail "failed to decode clouds.yaml" }}
+    {{ end -}}
+
+    {{- $$openstack := index $$clouds "clouds" "openstack" -}}
+
+    {{- if not (hasKey $$openstack "auth") }}
+      {{ fail "auth key not found in openstack config" }}
+    {{- end }}
+    {{- $$auth := index $$openstack "auth" -}}
+
+    {{- $$auth_url := index $$auth "auth_url" -}}
+    {{- $$app_cred_id := index $$auth "application_credential_id" -}}
+    {{- $$app_cred_name := index $$auth "application_credential_name" -}}
+    {{- $$app_cred_secret := index $$auth "application_credential_secret" -}}
+
+    {{- $$network_id := $$cluster.status.externalNetwork.id -}}
+    {{- $$network_name := $$cluster.status.externalNetwork.name -}}
+    ---
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: openstack-cloud-config
+      namespace: kube-system
+    type: Opaque
```

Comment:
would the already existing secret be updated with the strategic merge or would it be recreated? anyway i'd put the component label here explicitly

Reply:
same, this is managed/child cluster object

Reply:
it should be recreated by Sveltos from mgmt cluster

Reply:
the same answer as here https://github.com/k0rdent/kcm/pull/925#discussion_r1923603071

if was under the impression that these objects would persist within the management cluster, otherwise it's okay not to label

Reply:
resolving, keep one above for reference